pkg/broker/util.go

```diff
@@ -43,6 +43,82 @@ func toBrokerPlans(apbPlans []apb.Plan) []Plan {
 	return brokerPlans
 }
 
+func updateMetadata(metadata map[string]interface{}, params []apb.ParameterDescriptor) map[string]interface{} {
+	if params == nil || len(params) == 0 {
+		return metadata
+	}
+
+	if metadata == nil {
+		metadata = make(map[string]interface{})
+	}
+
+	formDefinition := make([]interface{}, 0)
+
+	for paramIdx := 0; paramIdx < len(params); {
+		var formItem interface{}
+		pd := params[paramIdx]
+		if pd.DisplayGroup == "" {
+			formItem, paramIdx = createUIFormItem(pd, paramIdx)
+		} else {
+			formItem, paramIdx = createUIFormGroup(params, pd.DisplayGroup, paramIdx)
+		}
+
+		formDefinition = append(formDefinition, formItem)
+	}
+
+	metadata["schemas"] = map[string]interface{}{
+		"service_instance": map[string]interface{}{
+			"create": map[string]interface{}{
+				"form_definition": formDefinition,
+			},
+			"update": map[string]interface{}{},
+		},
+		"service_binding": map[string]interface{}{
+			"create": map[string]interface{}{
+				"form_definition": formDefinition,
+			},
+		},
+	}
+
+	return metadata
+}
+
+func createUIFormGroup(params []apb.ParameterDescriptor, groupName string, paramIndex int) (map[string]interface{}, int) {
+	var formItem interface{}
+	items := []interface{}{}
+
+	for paramIndex < len(params) {
+		pd := params[paramIndex]
+		if pd.DisplayGroup != groupName {
+			break
+		}
+
+		formItem, paramIndex = createUIFormItem(pd, paramIndex)
+		items = append(items, formItem)
+	}
+
+	group := make(map[string]interface{})
```

Comment:
Do we ever see adding to the group? if we do, I would prefer this becomes a struct. If it is always going to be two values like this, then I don't see a problem with it.